src/control_position.c

```diff
@@ -79,16 +86,24 @@ void control_position_update(control_position_t* cpos, uint32_t now_us) {
 		-shifted_vel_coast, shifted_vel_coast
 	);
 
+	float new_vel_output_unshifted = 0;
 	if (fabs(cpos->prop_out) > fabs(cpos->vel_output_unshifted) || cpos->prop_out * cpos->vel_output_unshifted < 0) {
-		cpos->vel_output_unshifted = linear_ramp_to(
+		new_vel_output_unshifted = linear_ramp_to(
 			cpos->vel_output_unshifted,
 			cpos->cfg->acceleration * dt,
 			cpos->prop_out
 		);
 	} else {
-		cpos->vel_output_unshifted = cpos->prop_out;
+		new_vel_output_unshifted = cpos->prop_out;
 	}
 
+	if (cpos->direction_changed && signf(new_vel_output_unshifted) != signf(cpos->vel_output_unshifted)) {
```

Comment:
I think C compilers are too dumb to see that `signf(new_vel_output_unshifted) != signf(cpos->vel_output_unshifted)` can be converted to `new_vel_output_unshifted * cpos->vel_output_unshifted < 0` to make it much faster, so maybe we should replace it manually